package TestDDF;

message Simple 
{
    required int32 a = 1;
}

message ScalarTypes 
{
    required float  Float = 1;
    required double Double = 2;
    required int32  Int32 = 3;
    required uint32 Uint32 = 4;
    required int64  Int64 = 5;
    required uint64 Uint64 = 6;
    required string String = 7;
}

message Simple01
{
    required uint32 x = 1;
    required uint32 y = 2;
}

message Simple01Repeated
{
    repeated Simple01 array = 1;
}

message Simple02Repeated
{
    repeated uint32 array = 1;
}

message StringRepeated
{
    repeated string array = 1;
}

message Mesh
{
    enum Primitive
    {
        TRIANGLES = 1;
        QUADS = 2;
    }

    required string    Name = 1;
    repeated float     Vertices = 2;
    repeated uint32    Indices = 3;
    required uint32    PrimitiveCount = 4;
    required Primitive PrimitiveType = 5;
}

message NestedArraySub2
{
    required uint32 a = 1;
}

message NestedArraySub1
{
    required uint32 b = 1;
    repeated NestedArraySub2 array2 = 2;
    required uint32 c = 3;
}

message NestedArray
{
    required uint32 d = 1;
    repeated NestedArraySub1 array1 = 2;
    required uint32 e = 3;
}

